src/main/java/com/harriet/shopiify/auth/service/ShippingAddressService.java

```diff
@@ -0,0 +1,48 @@
+package com.harriet.shopiify.auth.service;
+
+import com.harriet.shopiify.auth.dto.ShippingAddressDTO;
+import com.harriet.shopiify.auth.model.ShippingAddress;
+import com.harriet.shopiify.auth.repository.ShippingAddressRepository;
+import com.harriet.shopiify.auth.vo.ShippingAddressUpdateVO;
+import com.harriet.shopiify.auth.vo.ShippingAddressVO;
+import org.springframework.beans.BeanUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class ShippingAddressService {
+    private final ShippingAddressRepository shippingAddressRepository;
+    public ShippingAddressService (ShippingAddressRepository shippingAddressRepository){
+        this.shippingAddressRepository=shippingAddressRepository;
+    }
+    public Long addShippingAddress (ShippingAddressVO vo){
+        ShippingAddress entity = new ShippingAddress();
+        entity.setShippingAddress(vo.getShippingAddress());
+        entity.setUserId(vo.getUserId());
+        return shippingAddressRepository.save(entity).getId();
+    }
+
+    public void updateShippingAddress(Long id, ShippingAddressUpdateVO vo){
+        ShippingAddress addressToUpdate = shippingAddressRepository.findById(id).get();
+        addressToUpdate.setShippingAddress(vo.getShippingAddress());
+        shippingAddressRepository.save(addressToUpdate);
+    }
+
+    public void deleteShippingAddress (long id){
+        shippingAddressRepository.deleteById(id);
+    }
+
+    public List<ShippingAddressDTO> findShippingAddByUserId ( Long userId){
```

Comment:
There's no need to use short form for any words in the method name as the IDE autocompletes them for you.

Reply:
which particular short-formed word are you referring to?

Reply:
`findShippingAddByUserId` should be `findShippingAddressByUserId `